arangod/Replication/ReplicationApplier.cpp

```diff
@@ -48,12 +48,13 @@ struct ApplierThread : public Thread {
     TRI_ASSERT(_syncer);
   }
 
-  ~ApplierThread() {
+  ~ApplierThread() {    
+    shutdown();
+
     {
       MUTEX_LOCKER(locker, _syncerMutex);
       _syncer.reset();
```

Comment:
As already discussed via Slack - the fact that a mutex is required in the destructor is per se a sign that there is a fundamental design problem: *it should never be possible for a destructor to be executed on an object that is currently being used by some other thread.*

But this has to be addressed in a separate PR.
Anyway, resetting `_syncer` only after the thread has shut down should put out the fire for now.